src/submodule.c

```diff
@@ -188,8 +188,7 @@ static int load_submodule_names(git_strmap *out, git_config *cfg)
 		git_buf_put(&buf, fdot + 1, ldot - fdot - 1);
 		git_strmap_insert(out, entry->value, git_buf_detach(&buf), rval);
 		if (rval < 0) {
```

Comment:
Not related to your change, which looks good to me, but is there any reason why `git_strmap_insert` is implemented as a macro? `rval` being used as a return value when it is at first sight passed in as an `int` only looks really horrible.

I'd at a minimum like to change it so the macro expects a pointer to an `int` so that the call here at least behaves somewhat like a normal function call.